jdaviz/core/template_mixin.py

```diff
@@ -230,32 +296,50 @@ def _update_has_subregions(self):
     def labels(self):
         return [s['label'] for s in self.items if 'label' in s.keys()]
 
+    @cached_property
+    def selected_item(self):
+        for item in self.items:
+            if item['label'] == self.selected:
+                return item
+        return {}
+
     @cached_property
     def selected_obj(self):
-        if self.selected == "Entire Spectrum":
+        if self.selected == self._default_text:
             return None
-        return self.app.get_subsets_from_viewer("spectrum-viewer",
-                                                subset_type="spectral").get(self.selected)
+        subset_type = self.selected_item['type']
+        # NOTE: this might cause some issues with multiple viewers in imviz which
```

Comment:
Didn't you say you found a workaround? Is this note still relevant?

Reply:
I updated the comment to explain the situation a little better (it shouldn't cause issues for imviz so long as all viewers share the same subsets, but in theory could cause us issues in the future for dynamically created viewers).